cli/lsp/tsc.rs

```diff
@@ -1678,20 +1678,28 @@ impl CompletionEntry {
   }
 
   fn get_filter_text(&self) -> Option<String> {
-    // TODO(@kitsonk) this is actually quite a bit more complex.
-    // See `MyCompletionItem.getFilterText` in vscode completion.ts.
-    if self.name.starts_with('#') && self.insert_text.is_none() {
-      return Some(self.name.clone());
+    if self.name.starts_with('#') {
+      if let Some(insert_text) = &self.insert_text {
+        if insert_text.starts_with("this.#") {
+          return Some(insert_text.replace("this.#", ""));
+        } else {
```

Comment:
Do we need to remove `this.` in the case where it only starts with `this.`?

Reply:
It is a replication of the vscode logic, which does not do this. And it only sets the filter text. When the replacement text is `this.#` there is then effectively empty filter text, which is then nothing.